test/strings/basic.jl

```diff
@@ -428,6 +428,9 @@ foobaz(ch) = reinterpret(Char, typemax(UInt32))
 # issue #17624, missing getindex method for String
 @test "abc"[:] == "abc"
 
-# PR #18259 reverted because failing test below
-a = [x for x in String([0xcf, 0x83, 0x83, 0x83, 0x83])]
-@test a[1] == 'σ'
+# we assume for performance that next/nextind return values past the end of a
+# string's underlying data; this helps with performance of `done`.
```

Comment:
The new comment is still kind of misleading, since we "rely" on something that the current implementation doesn't guarantee for invalid UTF-8. It should rather mention that it only works for valid strings.